### DISCUSSION TITLE
Is it Allowed to Inherit Workflows from an Organization's `.github` Repository?

### DISCUSSION THREAD

**Discussion by KrLite** [community/community#116846] [Actions] [answered]
### Select Topic Area

Question

### Body

I have a `.github` repository in an organization, with workflow files under `(repository root)/.github/workflows/` stored as `yml`. I remember that the doc says inheriting workflows from `.github` repositories is allowed, but I cannot find the reference. I end up finding no available workflows in my organization repositories, so I want to know that if it is really allowed to inherit workflows.

[This documentation](https://docs.github.com/en/actions/using-workflows/sharing-workflows-secrets-and-runners-with-your-organization#overview) is relevant but very unclear.

**Comment by larmitage-bjss** [comment, marked as answer]
(I am not an employee of github, just a fellow user with a bit of experience of this)

When the documentation says 'sharing workflows within your organisation' they mean using [reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) - this is the same as any other workflow but is triggered by `workflow_call`. You can pass inputs and outputs to/from this workflow similar to how actions work.

You still need a workflow in each repository which dictates the triggers and calls the reusable workflow that lives in some shared repository within your organisation - this shared repo just needs to make sure it has the right permissions set if it is not public - see [here](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#allowing-access-to-components-in-a-private-repository)

You can create a starter workflow as a template to make it easier to create this workflow in each repo (and/or add it to a template that's used whenever you create a new repo if needed). Starter workflows live in your special `.github` repository - see more info [here](https://docs.github.com/en/actions/using-workflows/creating-starter-workflows-for-your-organization)

**Reply by KrLite** [reply]
Thanks! I just figured that out too. Using reusable workflows is the solution, although it cannot inherit workflows directly (still needs workflow files declared in every branch), it can indeed reduce the maintenance complexity by gathering workflow calls together. And starter workflows are useful to manage caller workflows, too.

By the way, I encountered some misbehaves in `workflow_call`, and finally found that any of the secrets used by the called workflow must be explicitly passed in by the caller workflow to work, otherwise it fails, or just produces unexpected results. The easier solution is to add `secrets: inherit` in each caller workflow.

**Reply by larmitage-bjss** [reply]
You can point to workflows in different repositories and use a full path to reference workflows in the same repo but on a different branch/tag/sha if you need to make changes
e.g.
```
jobs:
  job1:
    uses: my-org/my-repo/.github/workflows/my-workflow.yml@some-branch-or-tag
```

**Reply by suicidemc** [reply]
TOP LINK MILHIOO)(
